### ISSUE TITLE
Enhance add option "Ignore Java text block" to Spell Checker

### ISSUE THREAD

**Issue by illustrare-opera** [eclipse-jdt/eclipse.jdt.ui#707]
In the current Eclipse IDE 2023‑06 Java code editor
with spell checking enabled and option "Ignore Java string literals" ticked I get:

```
String jdbcString = "jdbc"; // spelling ignored

String jdbcBlock = """
    jdbc""";                // tagged as misspelled
```
Add option to ignore spelling in Java text block.

**Comment by illustrare-opera** [comment]
Hi, it turns out to be fairly simple to add this enhancement only four files need patches.
Unfortunatly I lack the infrastructure to manage a pull request. Hoping that someone might have the time to have a look. The following patches successfully enable the requested enhancement in my R202306 Eclipse Java IDE.

```
org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/spelling/JavaSpellingEngine.java

 	protected void check(IDocument document, IRegion[] regions, ISpellChecker checker, ISpellingProblemCollector collector, IProgressMonitor monitor) {
 		SpellEventListener listener= new SpellEventListener(collector, document);
 		boolean isIgnoringJavaStrings= PreferenceConstants.getPreferenceStore().getBoolean(PreferenceConstants.SPELLING_IGNORE_JAVA_STRINGS);
+		boolean isIgnoringJavaMultiLineStrings= PreferenceConstants.getPreferenceStore().getBoolean(PreferenceConstants.SPELLING_IGNORE_JAVA_MULTI_LINE_STRINGS);
 		try {
 			for (IRegion region : regions) {
 				for (ITypedRegion partition : TextUtilities.computePartitioning(document, IJavaPartitions.JAVA_PARTITIONING, region.getOffset(), region.getLength(), false)) {
 					final String type= partition.getType();
 					if (isIgnoringJavaStrings && IJavaPartitions.JAVA_STRING.equals(type))
 						continue;
+					if (isIgnoringJavaMultiLineStrings && IJavaPartitions.JAVA_MULTI_LINE_STRING.equals(type))
+						continue;
 					if (!IDocument.DEFAULT_CONTENT_TYPE.equals(type) && !IJavaPartitions.JAVA_CHARACTER.equals(type))
 						checker.execute(listener, new SpellCheckIterator(document, partition, checker.getLocale(), monitor));
 				}
 				
 				
org.eclipse.jdt.ui/ui/org/eclipse/jdt/ui/PreferenceConstants.java
 
+ 	/**
+	 * A named preference that controls whether Java multiline strings
+	 * should be ignored during spell checking.
+	 * <p>
+	 * Value is of type <code>Boolean</code>.
+	 * </p>
+	 *
+	 * @since 3.30
+	 */
+	public static final String SPELLING_IGNORE_JAVA_MULTI_LINE_STRINGS= "spelling_ignore_java_multi_line_strings"; //$NON-NLS-1$;

 	 
 	 
org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/preferences/PreferencesMessages.java

+	public static String SpellingPreferencePage_ignore_java_multi_line_strings_label;

 	
 	
 	
org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/preferences/PreferencesMessages.properties

+SpellingPreferencePage_ignore_java_multi_line_strings_label=Ignore Java &text block literals

 
 
/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/preferences/SpellingConfigurationBlock.java

+	private static final Key PREF_SPELLING_IGNORE_JAVA_MULTI_LINE_STRINGS= getJDTUIKey(PreferenceConstants.SPELLING_IGNORE_JAVA_MULTI_LINE_STRINGS);

 
+		label= PreferencesMessages.SpellingPreferencePage_ignore_java_multi_line_strings_label;
+		slave= addCheckBox(user, label, PREF_SPELLING_IGNORE_JAVA_MULTI_LINE_STRINGS, trueFalse, 0);
+		allControls.add(slave);
+
 
 	private static Key[] getAllKeys() {
 		if (SUPPORT_CONTENT_ASSIST_PROPOSALS)
-			return new Key[] { PREF_SPELLING_USER_DICTIONARY, PREF_SPELLING_USER_DICTIONARY_ENCODING, PREF_SPELLING_IGNORE_DIGITS, PREF_SPELLING_IGNORE_MIXED, PREF_SPELLING_IGNORE_SENTENCE, PREF_SPELLING_IGNORE_UPPER, PREF_SPELLING_IGNORE_URLS, PREF_SPELLING_IGNORE_AMPERSAND_IN_PROPERTIES, PREF_SPELLING_IGNORE_NON_LETTERS, PREF_SPELLING_IGNORE_SINGLE_LETTERS, PREF_SPELLING_LOCALE, PREF_SPELLING_PROPOSAL_THRESHOLD, PREF_SPELLING_PROBLEMS_THRESHOLD, PREF_SPELLING_ENABLE_CONTENTASSIST, PREF_SPELLING_IGNORE_JAVA_STRINGS };
+			return new Key[] { PREF_SPELLING_USER_DICTIONARY, PREF_SPELLING_USER_DICTIONARY_ENCODING, PREF_SPELLING_IGNORE_DIGITS, PREF_SPELLING_IGNORE_MIXED, PREF_SPELLING_IGNORE_SENTENCE, PREF_SPELLING_IGNORE_UPPER, PREF_SPELLING_IGNORE_URLS, PREF_SPELLING_IGNORE_AMPERSAND_IN_PROPERTIES, PREF_SPELLING_IGNORE_NON_LETTERS, PREF_SPELLING_IGNORE_SINGLE_LETTERS, PREF_SPELLING_LOCALE, PREF_SPELLING_PROPOSAL_THRESHOLD, PREF_SPELLING_PROBLEMS_THRESHOLD, PREF_SPELLING_ENABLE_CONTENTASSIST, PREF_SPELLING_IGNORE_JAVA_STRINGS, PREF_SPELLING_IGNORE_JAVA_MULTI_LINE_STRINGS };
 		else
-			return new Key[] { PREF_SPELLING_USER_DICTIONARY, PREF_SPELLING_USER_DICTIONARY_ENCODING, PREF_SPELLING_IGNORE_DIGITS, PREF_SPELLING_IGNORE_MIXED, PREF_SPELLING_IGNORE_SENTENCE, PREF_SPELLING_IGNORE_UPPER, PREF_SPELLING_IGNORE_URLS, PREF_SPELLING_IGNORE_AMPERSAND_IN_PROPERTIES, PREF_SPELLING_IGNORE_NON_LETTERS, PREF_SPELLING_IGNORE_SINGLE_LETTERS, PREF_SPELLING_LOCALE, PREF_SPELLING_PROPOSAL_THRESHOLD, PREF_SPELLING_PROBLEMS_THRESHOLD, PREF_SPELLING_IGNORE_JAVA_STRINGS };
+			return new Key[] { PREF_SPELLING_USER_DICTIONARY, PREF_SPELLING_USER_DICTIONARY_ENCODING, PREF_SPELLING_IGNORE_DIGITS, PREF_SPELLING_IGNORE_MIXED, PREF_SPELLING_IGNORE_SENTENCE, PREF_SPELLING_IGNORE_UPPER, PREF_SPELLING_IGNORE_URLS, PREF_SPELLING_IGNORE_AMPERSAND_IN_PROPERTIES, PREF_SPELLING_IGNORE_NON_LETTERS, PREF_SPELLING_IGNORE_SINGLE_LETTERS, PREF_SPELLING_LOCALE, PREF_SPELLING_PROPOSAL_THRESHOLD, PREF_SPELLING_PROBLEMS_THRESHOLD, PREF_SPELLING_IGNORE_JAVA_STRINGS, PREF_SPELLING_IGNORE_JAVA_MULTI_LINE_STRINGS };
 	}

```

**Comment by jjohnstn** [comment]
Hi @illustrare-opera Thanks for the report and potential fix.  Actually, a new option is not needed.  A text block is a Java String and so the original ignore option should apply in that case as well so just the check needs to be added.  I can easily create a patch.  It may have to wait until 4.30 M1 to be merged as the 4.29 deadline for minor issues has passed.

**Comment by jjohnstn** [comment]
Closing this as it has been replaced by #771  Thanks @illustrare-opera for pointing this bug out.
